src/executors.jl

```diff
@@ -373,15 +381,26 @@ function dispatch!(exec::Executor, ctx::DispatchContext; throw_error=true)
     for more details.
     ```
     =#
-    f = Base.wrap_on_error(
-        Base.wrap_retry(
-            run_inner!,
-            allow_retry(retry_on(exec)),
-            retries(exec),
-            Base.DEFAULT_RETRY_MAX_DELAY
-        ),
-        on_error_inner!
-    )
+    @static if VERSION < v"0.6.0-dev.2042"
```

Comment:
Yeah, that seems fine. It's annoying that the ordering of the arguments changed as well, but I think this clearly identifies what's actually different.